compiler/nimlexbase.nim

```diff
@@ -99,16 +105,17 @@ proc fillBuffer(L: var TBaseLexer) =
       else:
         # rather than to give up here because the line is too long,
         # double the buffer's size and try again:
-        oldBufLen = L.buf.len
-        L.buf.setLen(L.buf.len * 2)
-        assert(L.buf.len - oldBufLen == oldBufLen)
+        oldBufLen = L.bufLen
+        L.bufLen = L.bufLen * 2
+        L.buf = cast[cstring](realloc(L.buf, L.bufLen * chrSize))
```

Comment:
why advantage does a manually managed `(cstring,len)` have over buf in this case, where we can reuse a string buffer?
is this for allowing slicing (does't look like it, and there's a better way) or for other reason?